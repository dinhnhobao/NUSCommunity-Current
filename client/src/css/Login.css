.login {
  margin: 0 auto;
  padding-bottom: 38px;
  width: 500px;
  border-radius: 20px;
  box-shadow: 0 3px 20px 0px rgba(0, 0, 0, 0.1);
}
.login-title {
  border-top-left-radius: 20px;
  border-top-right-radius: 20px;
  font-size: 30px;
  color: #fff;
  line-height: 126px;
  text-align: center;
  width: 100%;
  background-color: #35d0ba;
  font-weight: bolder;
}
.login-con {
  width: 390px;
  margin: 0 auto;
}
.login-con-img {
  width: 117px;
  display: block;
  margin: 10px auto;
}
.login-con-line {
  border-top: 1px solid rgba(0, 0, 0, 0.1);
  margin-bottom: 10px;
}
.login-con-form .ant-input {
  font-size: 16px;
  color: #1b3815;
  line-height: 20px;
  height: 55px;
  background: #ebebeb;
  border-radius: 27px;
  padding: 0 35px;
}
.login-con-form-reset {
  font-size: 16px;
  line-height: 16px;
  color: #e84a5f;
  padding-left: 20px;
}
.login-con-form-btn {
  background-color: #35d0ba;
  border-color: #35d0ba;
  border-radius: 50px;
  width: 100%;
  height: 50px;
  font-weight: bolder;
}
.login-con-form-btn:hover {
  background-color: #1b3815;
  border-color: #1b3815;
}
.login-con-form-btn:focus {
  background-color: #1b3815;
  border-color: #1b3815;
}
.login-con-form-register {
  color: #35d0ba;
  font-weight: bolder;
  line-height: 24px;
  margin-top: 38px;
  text-align: center;
  display: block;
}
