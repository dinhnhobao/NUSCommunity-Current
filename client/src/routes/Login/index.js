import React from "react";
import { Link } from "react-router-dom";
import Router from 'next/router'
import { Form, Input, Button, message} from "antd";
import icon from "../../assets/favicon.ico";
import "./index.css";
import Session from '../../utils/session';

// const onFinish = (values) => {
//   console.log(values);
//   message.success("login success~");
//   window.location.href = "/home";
// };

class Login extends React.Component {
  static async getInitialProps({req, res}) {
    let props = {
      session: ''
    }
    
    if (req && req.session) {
      props.session = req.session
    } else {
      props.session = await Session.getSession()
    }

    if (props.session && props.session.loggedin) {
      if (req) {
        res.redirect('/')
      } else {
        Router.push('/')
      }
    }
    
    return props
  }

  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: '',
      message: null
    }
    this.handleEmailChange = this.handleEmailChange.bind(this)
    this.handlePasswordChange = this.handlePasswordChange.bind(this)
    this.handleLogin = this.handleLogin.bind(this)
  }

  handleEmailChange(event) {
    this.setState({
      email: event.target.value.trim()
    })
  }

  
  handlePasswordChange(event) {
    this.setState({
      password: event.target.value.trim()
    })
  }
  
  handleLogin(event) {
    event.preventDefault()

    this.setState({
      message: null
    })

    if (!this.state.email || !this.state.password) {
      this.setState({
        message: 'Email/Password is empty!'
      })

      return
    }

    let data = {
      email: this.state.email,
      password: this.state.password
    }

    fetch('auth/login', {
      method: 'POST',
      body: JSON.stringify(data),
      headers:{
        'Content-Type': 'application/json'
      }
    })
    .then(res => res.json())
    .then(response => {
      if (response.loggedin) {
        //message.success("login success~");
        //window.location.href = "/index";
        const { history } = this.props;
        message.success("login success~");
        history.push("/index");
        //response.redirect('/home');
        //Router.push(`/`)
      } else if (response.message) {

        this.setState({
           message: response.message
        })
        //alert(response.message);
      } else {
        this.setState({
          message: 'Unknown Error!'
        })
      }
    })
    .catch(error => {
      console.error('Error:', error)
      this.setState({
        message: 'Request Failed!'
      })
    })
  }

  render() {
      const alert = (this.state.message === null) ? <div/> : <div className="alert-style" role="alert">{this.state.message}</div>
      return (
        <div style={{ paddingTop: 25 }}>
          <div className="login">
            <div className="login-title">Welcome to the NUSCommunity !</div>
            <div className="login-con">
              <img className="login-con-img" src={icon} alt="" />
              <div className="login-con-line" />
              <Form
                name="normal_login"
                className="login-con-form"
                // onFinish={this.onFinish}
              >
                <Form.Item
                  name="email"
                  rules={[
                    {
                      required: true,
                      message: "Please enter your email!",
                    },
                  ]}
                >
                  <Input placeholder="Email" name="email" value={this.state.email} onChange={this.handleEmailChange}/>
                </Form.Item>
                <Form.Item
                  name="password"
                  rules={[
                    {
                      required: true,
                      message: "Please input your password!",
                    },
                  ]}
                >
                  <Input type="password" name="password" placeholder="Password" value={this.state.password} onChange={this.handlePasswordChange}/>
                </Form.Item>
                <Form.Item>
                  <Link className="login-con-form-reset" to="/reset">
                    forgot password?
                  </Link>
                </Form.Item>
                
                <Form.Item>
                  <Button
                    type="submit"
                    onClick={this.handleLogin}
                    className="login-con-form-btn"
                    size="large"
                  >
                    SIGN IN
                  </Button>
                  {alert}
                </Form.Item>
                <Link className="login-con-form-register" to="/reg">
                  REGISTER HERE
                </Link>
              </Form>
            </div>
          </div>
        </div>
      );
  };
}
export default Login;
