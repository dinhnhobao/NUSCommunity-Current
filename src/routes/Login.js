import React from "react";
import { Link } from "react-router-dom";
import { Form, Input, Button, message } from "antd";
import icon from "../assets/favicon.ico";
import "../css/Login.css";

const Login = () => {
  const onFinish = (values) => {
    console.log(values);
    message.success("login success~");
    window.location.href = "/home";
  };

  return (
    <div style={{ paddingTop: 25 }}>
      <div className="login">
        <div className="login-title">Welcome to the NUSCommunity !</div>
        <div className="login-con">
          <img className="login-con-img" src={icon} alt="" />
          <div className="login-con-line" />
          <Form
            name="normal_login"
            className="login-con-form"
            onFinish={onFinish}
          >
            <Form.Item
              name="email"
              rules={[
                {
                  required: true,
                  message: "Please enter your email!",
                },
              ]}
            >
              <Input placeholder="Email" />
            </Form.Item>
            <Form.Item
              name="password"
              rules={[
                {
                  required: true,
                  message: "Please input your password!",
                },
              ]}
            >
              <Input type="password" placeholder="Password" />
            </Form.Item>
            <Form.Item>
              <Link className="login-con-form-reset" to="/reset">
                forgot password?
              </Link>
            </Form.Item>
            <Form.Item>
              <Button
                htmlType="submit"
                type="primary"
                className="login-con-form-btn"
                size="large"
              >
                SIGN IN
              </Button>
            </Form.Item>
            <Link className="login-con-form-register" to="/reg">
              REGISTER HERE
            </Link>
          </Form>
        </div>
      </div>
    </div>
  );
};
export default Login;
