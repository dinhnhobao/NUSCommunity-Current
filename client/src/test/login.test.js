import React from 'react';
import {shallow} from 'enzyme';
import Login from '../routes/Login/index';

const setUp = (props={}) => {
    const component = shallow(<Login {...props} />);
    return component;
}

const findByTestAttr = (component, attr) => {
    const wrapper = component.find(`[data-test='${attr}']`);
    return wrapper;
}

describe('Login Component', () => {
    let component;
    beforeEach(() => {
        component = setUp();
    })

    it('Should render correctly', () => {
        const wrapper =  findByTestAttr(component, 'loginComponent');
        expect(wrapper).toMatchSnapshot();
    });

    it('Should render without errors', () => {
        const wrapper =  findByTestAttr(component, 'loginComponent');
        expect(wrapper.length).toEqual(1);       
    });

    describe('Form', () => {
        it('Fill out email in the form', () => {
            const wrapper = shallow(<Login/>);
            let emailInput = wrapper.find('Input').first();
            emailInput.simulate('change', { 
                target: {value: 'e0407663'},
            })
            emailInput = wrapper.find('Input').first();
            expect(emailInput.props().value).toEqual('e0407663')
        })

        it('Fill out password in the form', () => {
            const wrapper = shallow(<Login/>);
            let passwordInput = wrapper.find('Input').last();
            passwordInput.simulate('change', { 
                target: {value: 'Anonymous'},
            })
            passwordInput = wrapper.find('Input').last();
            expect(passwordInput.props().value).toEqual('Anonymous')
        })
    })

    describe('Button', () => {
        it('Should render Button with click handle', () => {
            const wrapper =  findByTestAttr(component, 'buttonComponent');
            expect(wrapper.length).toEqual(1);
        })
    })
})
