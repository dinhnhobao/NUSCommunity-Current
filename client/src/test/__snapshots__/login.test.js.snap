// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component Should render correctly 1`] = `ShallowWrapper {}`;
